index.js

```diff
@@ -85,14 +85,36 @@ function checkMessagePathQueryParameter(req, res, next) {
 }
 
 function checkTopicQueryParameter(req, res, next) {
-
     if (req.query.topic) {
         req.body.topic = req.query.topic;
     }
+    next();
+}
 
+function checkQoSParameter(req,res,next){
+  if (req.body.qos){
```

Comment:
Please update to if (!req.body.qos){, otherwise you will always reset qos parameter.